process/block/interceptedBlocks/interceptedEquivalentProof.go

```diff
@@ -194,6 +201,21 @@ func (iep *interceptedEquivalentProof) CheckValidity() error {
 		return common.ErrAlreadyExistingEquivalentProof
 	}
 
+	marshalledProof, errNotCritical := iep.marshaller.Marshal(iep.proof)
+	if errNotCritical != nil {
+		logging.LogErrAsWarnExceptAsDebugIfClosingError(log, errNotCritical,
+			"CheckValidity.Marshal proof",
+			"err", errNotCritical)
+		return nil
+	}
+
+	errNotCritical = iep.store.Put(dataRetriever.ProofsUnit, iep.proof.GetHeaderHash(), marshalledProof)
```

Comment:
we save headers to storage only on CommitBlock? should we do the same with proofs? i'm thinking in case of reverts/rollbacks

Reply:
same applies to the pool.. we should see the testing results, already discussed this with @AdoAdoAdo